functions/src/functions/helpers.ts

```diff
@@ -16,28 +16,28 @@ import {
   onRequest,
   type Request,
 } from 'firebase-functions/v2/https'
-import { type TypeOf, z, type ZodType } from 'zod'
+import { z } from 'zod'
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
 /* eslint-disable @typescript-eslint/no-redundant-type-constituents */
 
 export function validatedOnCall<
-  Schema extends ZodType<any, any, any>,
+  Schema extends z.ZodTypeAny,
   Return = any | Promise<any>,
 >(
   name: string,
   schema: Schema,
   handler: (request: CallableRequest<z.output<Schema>>) => Return,
 ): CallableFunction<
-  z.output<Schema>,
+  z.input<Schema>,
```

Comment:
We need to use `z.input` here, because Schema can parse `z.input` to `z.output`. `handler` receives `z.output` correctly.